### ISSUE TITLE
BN crashes when reanalyzing an older BNDB

### ISSUE THREAD

**Issue by alexrp** [Vector35/binaryninja-api#4359]
**Version and Platform (required):**
- Binary Ninja Version: 3.5.4307-dev Personal (390387c6)
- OS: Windows
- OS Version: 10.0.25375.1
- CPU Architecture: x64

**Bug Description:**
```
[0x0]   binaryninjacore!BNParseExpression + 0x203180   
[0x1]   binaryninjacore!BNParseExpression + 0x203dec   
[0x2]   binaryninjacore!BNParseExpression + 0x1a7a99   
[0x3]   binaryninjacore!BNUpdateHighLevelILOperand + 0x7cd69   
[0x4]   binaryninjacore!BNUpdateHighLevelILOperand + 0x7d04c   
[0x5]   binaryninjacore!BNUpdateHighLevelILOperand + 0x7fb34   
[0x6]   binaryninjacore!BNRegisterGlobalFunctionRecognizer + 0x8b2cd   
[0x7]   binaryninjacore!BNRegisterGlobalFunctionRecognizer + 0xa2ab4   
[0x8]   binaryninjacore!BNSetFlowGraphNodeLines + 0x1356c5   
[0x9]   binaryninjacore!BNWriteWebsocketClientData + 0x85d2   
[0xa]   binaryninjacore!BNWriteWebsocketClientData + 0x3361   
[0xb]   ucrtbase!thread_start<unsigned int (__cdecl*)(void *),1> + 0x31   
[0xc]   KERNEL32!BaseThreadInitThunk + 0x21   
[0xd]   ntdll!RtlUserThreadStart + 0x21   
```

**Steps To Reproduce:**
Open [this](https://drive.google.com/file/d/1lWxswaAQ9GoY4W4gB6D52zrBUokZn3Ge/view?usp=sharing) BNDB and let reanalysis take place.

**Expected Behavior:**
No crash.

**Comment by negasora** [comment]
I'm not seeing a crash on `3.5.4309-dev` - does it still crash for you on that version?

**Comment by alexrp** [comment]
It reproduced consistently on `3.5.4307-dev`, but indeed appears to be resolved in `3.5.4309-dev`.
